### PR TITLE
71 scheduler tasks

### LINKED ISSUE #71: add scheduler jobs to database
add scheduler jobs to a new database collection to keep track of jobs.

### PR DESCRIPTION
**Pull request by karelumair**
add scheduler job details to the new database collection.

closes #71

### CODE REVIEW

**Review by pratikGhodke1 — APPROVED** [review]
Looks awesome! Can we have a demo?

**Comment by karelumair** [comment]
Added Status Enum and update Task Status

File: database/models.py [hunk 1]
```diff
@@ -71,3 +72,22 @@ def to_json(self, *args, **kwargs):
...
         return obj
+
+
+class TaskStatus(Enum):
+    """Possible Task Statuses"""
+
+    NOT_STARTED = {"quotes": "NOT_STARTED", "author": "NOT_STARTED"}
+    QUOTE_IN_PROGRESS = {"quotes": "IN_PROGRESS", "author": "NOT_STARTED"}
+    AUTHOR_IN_PROGRSS = {"quotes": "SUCCESS", "author": "IN_PROGRESS"}
+    SUCCESS = {"quotes": "SUCCESS", "author": "SUCCESS"}
```
Comment (pratikGhodke1) [inline database/models.py]: Enums should contain constants such as int, str and not the dictionary. You can use them like - 
```suggestion
class TaskStatus(Enum):
    """Task Statuses"""

    NOT_STARTED: str = "NOT_STARTED"
    IN_PROGRESS: str = "IN_PROGRESS"
    FAILED: str = "FAILED"
    SUCCESS: str = "SUCCESS"
```

Comment (karelumair) [reply database/models.py]: But then I can't store result as dict type.

File: database/models.py [hunk 2]
```diff
@@ -71,3 +72,22 @@ def to_json(self, *args, **kwargs):
...
+
+
+class ScheduledTask(db.Document):
+    """Scheduled Tasks Model"""
+
+    jobId = db.StringField()
+    celeryTask = db.StringField()
+    description = db.StringField()
+    status = db.DictField(default=TaskStatus.NOT_STARTED.value)
```
Comment (pratikGhodke1) [inline database/models.py]:
```suggestion
    status = db.DictField(default={"quotes": TaskStatus.NOT_STARTED, "authors": TaskStatus.NOT_STARTED})
```

File: database/models.py (hunk 1 above)
Comment (karelumair) [reply database/models.py]: I will need to use StringField

File: database/models.py (hunk 2 above)
Comment (pratikGhodke1) [reply database/models.py]: You can actually create a function which will populate this fields and returns above required dictionary with correct enum fields

File: database/models.py (hunk 1 above)
Comment (karelumair) [reply database/models.py]: can't store like you said '{quotes: SUCCESS, author: IN_PROGRESS}'

File: database/models.py (hunk 2 above)
Comment (karelumair) [reply database/models.py]: Ok got it.

**Comment by karelumair** [comment]
Done with all required changes.
Added dataclass to update task status

File: database/models.py [hunk 3]
```diff
@@ -71,3 +73,22 @@ def to_json(self, *args, **kwargs):
...
+
+
+class TaskStatus(Enum):
+    """Possible Task Statuses"""
+
+    NOT_STARTED: str = "NOT_STARTED"
+    IN_PROGRESS: str = "IN_PROGRESS"
+    FAILED: str = "FAILED"
+    SUCCESS: str = "SUCCESS"
+
```
Comment (pratikGhodke1) [inline database/models.py]: Add this in scraper utils file

File: utils/utils.py [hunk 4]
```diff
@@ -1,12 +1,37 @@
...
 from flask_mongoengine import BaseQuerySet
 from pymongo.command_cursor import CommandCursor
 
 
+@dataclass
+class ScrapingTask:
+    """Dataclass for Scraping Task Status"""
+
+    quote: str = "NOT_STARTED"
+    author: str = "NOT_STARTED"
```
Comment (pratikGhodke1) [inline utils/utils.py]: Use Enum values here.
```suggestion
    quote: str = TaskStatus.NOT_STARTED.value
    author: str = TaskStatus.NOT_STARTED.value
```

File: utils/utils.py [hunk 5]
```diff
@@ -1,12 +1,37 @@
...
+        if author:
+            self.author = author.value
+
+        self.task.update(status=self.to_dict())
+
+    def to_dict(self):
+        """Return dict of task status"""
+
+        return {"quote": self.quote, "author": self.author}
+
```
Comment (pratikGhodke1) [inline utils/utils.py]: Add typehints

**Comment by pratikGhodke1** [comment]
Move `utils/scraping.py (could rename it to scraper.py) and related functions` to a new folder `services/scraper` and access it from there. Make this changes in refactoring Issue. Just a hint here what can be done to arrange these files and functionalities.

**Pull request merged**